src/main/java/com/google/codeu/servlets/MessageServlet.java

```diff
@@ -78,7 +80,15 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
     String user = userService.getCurrentUser().getEmail();
     String text = Jsoup.clean(request.getParameter("text"), Whitelist.none());
 
-    Message message = new Message(user, text);
+    //replace user inputted text if it fits the defined patten into an img tag
+    String regex = "(https?://\\S+\\.(png|jpg))";
+    String replacement = "<img src=\"$1\" />";
+    String replacedText = text.replaceAll(regex, replacement);
+
+    //***need to add url validation below***
```

Comment:
A common format for these kind of comments is `TODO(dayhahm): Add URL validation here.` That way it's easy to track down who this comment is left for. Another common format is `TODO(issue #): Add URL validation here` where `issue #` corresponds to an entry in your project tracking software.